src/main/java/be/minelabs/block/ExtraDispenserBehavior.java

```diff
@@ -59,6 +62,23 @@ protected ItemStack dispenseSilently(BlockPointer pointer, ItemStack stack) {
                 return stack;
             }
         });
+
+        DispenserBlock.registerBehavior(Items.BALLOON, new ItemDispenserBehavior() {
+            protected ItemStack dispenseSilently(BlockPointer pointer, ItemStack stack) {
+                if(stack.getItem() instanceof BalloonItem bi) {
+                    BlockPos blockPos = pointer.getPos().offset((Direction)pointer.getBlockState().get(DispenserBlock.FACING));
```

Comment:
Unnecessary casting.
IntelliJ shows you "grayed" casts (meaning they are not unnecessary).